File: windows/LanDrop.Windows/Protocol.cs
namespace LanDrop.Windows;
public static class Protocol { public const int DiscoveryPort = 40404; public const int HttpPort = 40405; }
public sealed class Peer(string name, string platform, string address, int port, DateTime lastSeen)
{
    public string Name { get; set; } = name; public string Platform { get; } = platform; public string Address { get; } = address; public int Port { get; } = port; public DateTime LastSeen { get; set; } = lastSeen;
    public string Display => $"{Name} · {Platform}\n{Address}:{Port}";
}

File: android/app/src/main/java/com/maidehua/landrop/Protocol.kt
package com.maidehua.landrop

import kotlinx.serialization.Serializable

object Protocol { const val DISCOVERY_PORT = 40404; const val HTTP_PORT = 40405 }
@Serializable data class DiscoveryMessage(val app: String="LanDrop", val version: Int=1, val name: String, val platform: String="android", val port: Int=Protocol.HTTP_PORT)
data class Peer(val name: String, val platform: String, val address: String, val port: Int, val lastSeen: Long = System.currentTimeMillis()) { val display get()="$name · $platform\n$address:$port" }
